frontend/public/features.ts

```diff
@@ -82,27 +90,30 @@ const detectOpenShift = dispatch => coFetchJSON(openshiftPath)
       : handleError(err, FLAGS.OPENSHIFT, dispatch, detectOpenShift)
   );
 
-const detectCanListNS = dispatch => {
-  const req = {
-    spec: {
-      resourceAttributes: {
-        name: 'namespaces',
-        verb: 'list',
-      },
-    },
-  };
-  k8sCreate(SelfSubjectAccessReviewModel, req)
-    .then(
-      res => setFlag(dispatch, FLAGS.CAN_LIST_NS, res.status.allowed),
-      err => handleError(err, FLAGS.CAN_LIST_NS, dispatch, detectCanListNS)
-    );
+const detectSSARRestrictions = dispatch => {
+  [
+    [ FLAGS.CAN_LIST_NS,    { spec: { resourceAttributes: { name: 'namespaces', verb: 'list' } } }],                // eslint-disable-line no-multi-spaces
+    [ FLAGS.CAN_LIST_NODE,  { spec: { resourceAttributes: { name: 'nodes', verb: 'list' } } }],                     // eslint-disable-line no-multi-spaces
+    [ FLAGS.CAN_LIST_PVC,   { spec: { resourceAttributes: { name: 'persistentvolumeclaims', verb: 'list' } } }],    // eslint-disable-line no-multi-spaces
+    [ FLAGS.CAN_LIST_STORE, { spec: { resourceAttributes: { name: 'storageclasses', verb: 'list' } } }],            // eslint-disable-line no-multi-spaces
+    [ FLAGS.CAN_LIST_CRD,   { spec: { resourceAttributes: { name: 'customresourcedefinitions', verb: 'list' } } }], // eslint-disable-line no-multi-spaces
+  ].forEach(restriction => {
+    const FLAG = _.head(restriction);
+    const req = _.last(restriction);
+    k8sCreate(SelfSubjectAccessReviewModel, req)
+      .then(
+        (res) => setFlag(dispatch, FLAG, res.status.allowed),
+        (err) => handleError(err, FLAG, dispatch, detectSSARRestrictions)
```

Comment:
handleError may call detectSSARRestrictions which will go through the entire list again.

Reply:
Thanks for the catch!  Yup, fine when this was previously one resource, not awesome now.  I'll fix.

Reply:
Any concerns with this kind of approach:

```JavaScript 
const detectSSARFns = _.transform([
  [ FLAGS.CAN_LIST_NS, 'namespaces'],
  [ FLAGS.CAN_LIST_NODE, 'nodes'],
  [ FLAGS.CAN_LIST_PVC, 'persistentvolumeclaims'],
  [ FLAGS.CAN_LIST_STORE, 'storageclasses'],
  [ FLAGS.CAN_LIST_CRD, 'customresourcedefinitions'],
], (result, restriction) => {
  const FLAG = _.head(restriction);
  const req = { spec: { resourceAttributes: { name: _.last(restriction), verb: 'list' } } };
  const fnName = `detect${_.upperFirst(_.camelCase(FLAG))}`;
  result[fnName] = (dispatch) => {
    k8sCreate(SelfSubjectAccessReviewModel, req)
      .then(
        (res) => setFlag(dispatch, FLAG, res.status.allowed),
        (err) => handleError(err, FLAG, dispatch, fnName)
      );
  } 
}); 

export const featureActions = _.assign({
  detectSecurityLabellerFlags,
  detectCalicoFlags,
  detectOpenShift,
  detectSSARRestrictions,
}, detectSSARFns);
```
Reduces boilerplate, but its a little less intuitive what the functions are named in the end.  Its that vs just breaking this out into 5 copy/paste functions.

Reply:
Yeah, lets not copy paste.  Transforming the flags into function names is a bit brittle - maybe adding them to the list would be better.  Also, you can write the assign natively btw:

```javascript
const featureActions = {
  ...detectSSARFns,
  detectSecurityLabellerFlags,
  detectCalicoFlags,
  detectOpenShift,
  detectSSARRestrictions,
};

Reply:
Yeah, seems a little gross to make the function names as strings in the list, but that would eliminate the quirky transform & make it an easy read.  I'll update.